lib/pbio/src/image/image.c

```diff
@@ -262,6 +262,142 @@ void pbio_image_draw_vline(pbio_image_t *image, int x, int y, int l,
     }
 }
 
+/**
+ * Draw a line with a flat slope (less or equal to 1).
+ * @param [in,out] image      Image to draw into.
+ * @param [in]     x1         X coordinate of the first end.
+ * @param [in]     y1         Y coordinate of the first end.
+ * @param [in]     x2         X coordinate of the second end.
+ * @param [in]     y2         Y coordinate of the second end.
+ * @param [in]     value      Pixel value.
+ *
+ * This is an internal function, x2 must be greater or equal to x1.
+ *
+ * Clipping: drawing is clipped to image dimensions.
+ */
+static void pbio_image_draw_line_flat(pbio_image_t *image, int x1, int y1,
+    int x2, int y2, uint8_t value) {
+    int dx = x2 - x1;
+    int dy = y2 - y1;
+    int ydir = 1;
+    int x, y, err;
+
+    // Fall back to horizontal line, much faster.
+    if (dy == 0) {
+        pbio_image_draw_hline(image, x1, y1, dx + 1, value);
+        return;
+    }
+
+    // Clipping, X out of image.
+    if (x1 >= image->width || x2 < 0) {
+        return;
+    }
+
+    // Check Y direction.
+    if (dy < 0) {
+        dy = -dy;
+        ydir = -1;
+    }
+
+    // Error is scaled by 2 * dx, offset with one half to look at mid point.
+    err = -dx;
+    y = y1;
+
+    // Skip pixels left of image.
+    if (x1 < 0) {
+        err += -x1 * dy * 2;
+        int yskip = (err + dx * 2) / (dx * 2);
+        err -= yskip * dx * 2;
+        y += yskip * ydir;
+        x1 = 0;
+    }
+
+    // Skip pixels right of image.
+    if (x2 >= image->width) {
+        x2 = image->width - 1;
+    }
+
+    // Draw.
+    x = x1;
+    do {
+        pbio_image_draw_pixel(image, x, y, value);
+        err += dy * 2;
+        if (err >= 0) {
+            err -= dx * 2;
+            y += ydir;
+        }
+        x++;
+    } while (x <= x2);
+}
+
+/**
+ * Draw a line with a steep slope (more than 1).
+ * @param [in,out] image      Image to draw into.
```

Comment:
```suggestion
 * @param [in] image      Image to draw into.
```

And everywhere else too.

Reply:
Done.